src/components/Pagination/index.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import Select from '../Select';
+import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
+import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
+import PropTypes from 'prop-types';
+import PaginationElement from './PaginationElement';
+import { IconArrow, LastElement, PaginationContainer } from './Pagination.styled';
+
+const Pagination = ({ activePage, pageNum, showedAmount }) => {
+    const getElements = () => {
+        const lastShowedPage = showedAmount > pageNum ? pageNum  : showedAmount;
+        const newShowedPages = [];
+        for(let i = activePage; i <= lastShowedPage; i++) {
+            newShowedPages.push(i);
+        }
+        return newShowedPages;
+    }
+
+    const [showedPages, setShowedPages] = useState(getElements());
+    const [isHoverLeftArrow, setIsHoverLeftArrow] = useState(false);
+    const [isHoverRightArrow, setIsHoverRightArrow] = useState(false);
+
+    return(
+        <PaginationContainer>
+            <IconArrow 
```

Comment:
I would try to get rid of mouseOver states here and put to styled components. It would make the component much simpler